src/Http/Authenticator.php

```diff
@@ -97,6 +97,9 @@ public function refreshToken(): ?string
         return $this->refreshToken;
     }
 
+    /**
+     * @deprecated
+     */
```

Comment:
```suggestion
```

While I much prefer constructor injection rather than this kind of setter, it feels a bit arbitrary to deprecate this one at this point (when we're not yet making any changes about Guzzle), while this same class provides a few other similar setters. Makes sense? If we end up merging the PSR18 changes, then we can deprecate it 👍